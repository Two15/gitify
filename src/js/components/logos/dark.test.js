import React from 'react'; // eslint-disable-line no-unused-vars
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LogoDark from './dark';

describe('components/logos/dark.js', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LogoDark />);

    expect(tree).toMatchSnapshot();
  });

  it('should click on the logo', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<LogoDark onClick={onClick} />);

    expect(wrapper).toBeDefined();

    wrapper.find('svg').simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
