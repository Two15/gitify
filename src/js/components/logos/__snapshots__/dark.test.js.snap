// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/logos/dark.js renders correctly 1`] = `
<svg
  className="logo"
  onClick={[Function]}
  viewBox="0 0 180.31 182.52"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <linearGradient
      gradientUnits="userSpaceOnUse"
      id="linear-dark-gradient"
      x1="300.53"
      x2="130.22"
      y1="294.89"
      y2="240.53"
    >
      <stop
        offset="0"
        stopColor="#22283B"
      />
      <stop
        offset="1"
        stopColor="#555B6E"
      />
    </linearGradient>
    <linearGradient
      id="linear-dark-gradient-2"
      x1="315.88"
      x2="216.87"
      xlinkHref="#linear-dark-gradient"
      y1="246.16"
      y2="214.57"
    />
    <linearGradient
      id="linear-dark-gradient-3"
      x1="264.39"
      x2="147.04"
      xlinkHref="#linear-dark-gradient"
      y1="211.56"
      y2="174.1"
    />
  </defs>
  <path
    d="M178.67,226.51a47.7,47.7,0,0,0,89.93,21.91h46.56a91.61,91.61,0,0,1-88.82,69.35c-50.45,0-91.49-40.94-91.49-91.26C145,186.6,171,205,178.67,226.51Z"
    style={
      Object {
        "fill": "url(#linear-dark-gradient)",
      }
    }
    transform="translate(-134.85 -135.25)"
  />
  <path
    d="M214.16,226.51a21.91,21.91,0,0,1,21.91-21.91h38.57a12.08,12.08,0,0,1,8.3,4l32.21,39.86H236.07A21.91,21.91,0,0,1,214.16,226.51Z"
    style={
      Object {
        "fill": "url(#linear-dark-gradient-2)",
      }
    }
    transform="translate(-134.85 -135.25)"
  />
  <path
    d="M258.74,179.07H226.34a47.61,47.61,0,0,0-47.66,47.44c-11.27,24.3-30.12,41.63-43.83,0,0-50.32,41-91.26,91.49-91.26h32.41a21.91,21.91,0,0,1,0,43.83Z"
    style={
      Object {
        "fill": "url(#linear-dark-gradient-3)",
      }
    }
    transform="translate(-134.85 -135.25)"
  />
</svg>
`;
